### PR TITLE
Reduce default status check deadline to 2 mins

### PR DESCRIPTION
**Pull request by tejal29**
Related to #3638

In this PR, we decrease the default Status check deadline to 3 seconds. 

In this changes
- if `statusCheckDeadlineSeconds` is present in the skaffold deploy config, respect that.
- if not, then set deadline as maximum deadline seen in the `progressDeadlineSeconds` in `Deployment.Spec`
- If none of them specified, use `defaultStatusCheckDeadline`

### CODE REVIEW

File: pkg/skaffold/deploy/status_check.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ import (
 )
 
 var (
-	defaultStatusCheckDeadline = time.Duration(10) * time.Minute
+	defaultStatusCheckDeadline = time.Duration(2) * time.Minute
```
Comment (dgageot) [inline pkg/skaffold/deploy/status_check.go]: Can't you write `2 * time.Minute`?

File: pkg/skaffold/deploy/status_check.go [hunk 2]
```diff
@@ -110,7 +113,7 @@ func getDeployments(client kubernetes.Interface, ns string, l *DefaultLabeller,
 	deployments := make([]Resource, 0, len(deps.Items))
 	for _, d := range deps.Items {
 		var deadline time.Duration
-		if d.Spec.ProgressDeadlineSeconds == nil || *d.Spec.ProgressDeadlineSeconds > int32(deadlineDuration.Seconds()) {
+		if d.Spec.ProgressDeadlineSeconds == nil {
 			deadline = deadlineDuration
 		} else {
 			deadline = time.Duration(*d.Spec.ProgressDeadlineSeconds) * time.Second
```
Comment (dgageot) [inline pkg/skaffold/deploy/status_check.go]: Same here?

File: pkg/skaffold/deploy/status_check.go [hunk 3]
```diff
@@ -257,3 +260,16 @@ func (c *resourceCounter) markProcessed(err error) resourceCounter {
 		pods:        &podCp,
 	}
 }
+
+func statusCheckDeadline(value int, deployments []Resource) time.Duration {
+	if value > 0 {
+		return time.Duration(value) * time.Second
```
Comment (dgageot) [inline pkg/skaffold/deploy/status_check.go]: And here

File: pkg/skaffold/deploy/status_check.go [hunk 4]
```diff
@@ -257,3 +260,16 @@ func (c *resourceCounter) markProcessed(err error) resourceCounter {
...
+	}
+	d := time.Duration(0)
+	for _, r := range deployments {
+		if r.Deadline() > d {
+			d = r.Deadline()
```
Comment (dgageot) [inline pkg/skaffold/deploy/status_check.go]: `return r.Deadline()`

File: pkg/skaffold/deploy/status_check.go [hunk 5]
```diff
@@ -257,3 +260,16 @@ func (c *resourceCounter) markProcessed(err error) resourceCounter {
...
+		}
+	}
+	return d
```
Comment (dgageot) [inline pkg/skaffold/deploy/status_check.go]: `return time.Duration(0)`

File: pkg/skaffold/deploy/status_check.go (hunk 4 above)
Comment (tejal29) [reply pkg/skaffold/deploy/status_check.go]: I can't return here. i want to calculate the max deadline. I renamed the func to `statusCheckMaxDeadline`

File: pkg/skaffold/deploy/status_check.go (hunk 5 above)
Comment (tejal29) [reply pkg/skaffold/deploy/status_check.go]: no. this would return the max duration seen so far.

File: pkg/skaffold/deploy/status_check.go (hunk 2 above)
Comment (tejal29) [reply pkg/skaffold/deploy/status_check.go]: i dont understand what you mean?
Do you meant `*d.Spec.ProgressDeadline * time.Second`?
i see an IDE error
```
cannot use type int as type Duration in an assignment
```

and also a make error
```
pkg/skaffold/deploy/status_check.go:119:47: invalid operation: *d.Spec.ProgressDeadlineSeconds * time.Second (mismatched types int32 and time.Duration)

```

File: pkg/skaffold/deploy/status_check.go (hunk 3 above)
Comment (tejal29) [reply pkg/skaffold/deploy/status_check.go]: see above reply.

**Comment by tejal29** [comment]
@dgageot Addressed your comments. Please take another look.

File: pkg/skaffold/deploy/status_check.go (hunk 2 above)
Comment (dgageot) [reply pkg/skaffold/deploy/status_check.go]: My bad. Only works with constants

**Pull request merged**